go/inst/instance_dao.go

```diff
@@ -226,7 +226,14 @@ func logReadTopologyInstanceError(instanceKey *InstanceKey, hint string, err err
 // server and writes the result synchronously to the orchestrator
 // backend.
 func ReadTopologyInstance(instanceKey *InstanceKey) (*Instance, error) {
-	return ReadTopologyInstanceBufferable(instanceKey, false, nil)
+	instance, skipped, err := ReadTopologyInstanceBufferable(instanceKey, false, nil)
+	if skipped {
+		if config.Config.EnableDiscoveryFiltersLogs {
+			log.Infof("Skipping discovery of %+v because its replication user matches DiscoveryIgnoreReplicationUsernameFilters", instanceKey)
+		}
+		instance = nil
```

Comment:
in this case we'll return `nil` instance and also `nil` error, are we sure that it won't cause any panics?